From ea3fb6920bcc4903614886b2adae416299a5ac21 Mon Sep 17 00:00:00 2001
From: ye li <ye.li@nxp.com>
Date: Tue, 13 Sep 2016 21:00:47 +0800
Subject: [PATCH 1157/5242] MLK-12990 arm: dts: Fix CAAM JR interrupt parent
 issue on i.MX6SX

commit  8da9a292a1393588ac712d15f31e0f521fb46fa4 from
https://source.codeaurora.org/external/imx/linux-imx.git

To enable a IRQ as a wakeup source, we should set its interrupt parent to
gpc not the intc. Fix the wrong CAAM JR interrupts setting by removing the
its interrupt parent to use default SOC's interrupt parent "gpc". Otherwise,
irq_set_irq_wake will return error.

Signed-off-by: ye li <ye.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sx.dtsi |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index 8c8ffb6..3dd07c6 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -953,7 +953,6 @@
 				#size-cells = <1>;
 				reg = <0x2100000 0x40000>;
 				ranges = <0 0x2100000 0x40000>;
-				interrupt-parent = <&intc>;
 				clocks = <&clks IMX6SX_CLK_CAAM_MEM>,
 					 <&clks IMX6SX_CLK_CAAM_ACLK>,
 					 <&clks IMX6SX_CLK_CAAM_IPG>,
@@ -963,15 +962,13 @@
 				sec_jr0: jr0@1000 {
 					compatible = "fsl,sec-v4.0-job-ring";
 					reg = <0x1000 0x1000>;
-					interrupt-parent = <&intc>;
-					interrupts = <0 105 0x4>;
+					interrupts = <GIC_SPI 105 IRQ_TYPE_LEVEL_HIGH>;
 				};
 
 				sec_jr1: jr1@2000 {
 					compatible = "fsl,sec-v4.0-job-ring";
 					reg = <0x2000 0x1000>;
-					interrupt-parent = <&intc>;
-					interrupts = <0 106 0x4>;
+					interrupts = <GIC_SPI 106 IRQ_TYPE_LEVEL_HIGH>;
 				};
 			};
 
-- 
1.7.9.5

